test/phoenix/channel_test.exs

```diff
@@ -421,8 +421,7 @@ defmodule Phoenix.ChannelTest do
                        payload: fn socket -> {:ok, socket} end}
     assert {:ok, _sockets} = Transport.dispatch(message, sockets, self, Router, LongPoller)
     assert_received {:join, "topic2-override:somesubtopic"}
-    assert {:error, :bad_transport_match, _sockets} =
-      Transport.dispatch(message, sockets, self, Router, WebSocket)
-    refute_received {:join, "topic2-override:somesubtopic"}
+    assert {:ok, _sockets} = Transport.dispatch(message, sockets, self, Router, WebSocket)
+    assert_received {:join, "topic2-override:somesubtopic"}
```

Comment:
Reason why this (and the other above) were failing was because of the `:` match problem when calling `match_channel`. This test logically should do the same as `LongPoller`, which is a successful join.